### PR TITLE
add dataset subscript

### PR DESCRIPTION
**Pull request by nevermore3**

### CODE REVIEW

File: src/common/expression/SubscriptExpression.cpp [hunk 1]
```diff
@@ -41,6 +41,19 @@ const Value& SubscriptExpression::eval(ExpressionContext &ctx) {
             result_ = lvalue.getMap().at(rvalue.getStr());
             break;
         }
+        if (lvalue.isDataSet()) {
```
Comment (dutor) [inline src/common/expression/SubscriptExpression.cpp:44]: Could you please add some UTs about `DataSet[]` and `DataSet[][]`?

**Review by dutor — COMMENTED** [review]
Well done.

File: src/common/expression/SubscriptExpression.cpp [hunk 2]
```diff
@@ -41,6 +41,19 @@ const Value& SubscriptExpression::eval(ExpressionContext &ctx) {
...
+            if (!rvalue.isInt()) {
+                break;
+            }
+            auto size = static_cast<int64_t>(lvalue.getDataSet().size());
```
Comment (dutor) [inline src/common/expression/SubscriptExpression.cpp:48]: Please DO compile before every submit and update.

File: src/common/expression/test/ExpressionTest.cpp [hunk 3]
```diff
@@ -1393,6 +1393,85 @@ TEST_F(ExpressionTest, NotInMap) {
...
+            std::vector<Value> val;
+            val.reserve(5);
+            for (int32_t j = 0; j < 5; j++) {
+                val.emplace_back(i + j);
+            }
+            ds.rows.emplace_back(List(std::move(val)));
+        }
+        auto *dataset = new ConstantExpression(ds);
+        auto *rowIndex = new ConstantExpression(0);
+        auto *colIndex = new ConstantExpression(5);
```
Comment (Shylock-Hg) [inline src/common/expression/test/ExpressionTest.cpp:1468]: Shall we support string column index?

**Pull request merged**